asg/src/expression/circuit_init.rs

```diff
@@ -165,7 +165,7 @@ impl<'a> Into<leo_ast::CircuitInitExpression> for &CircuitInitExpression<'a> {
             members: self
                 .values
                 .iter()
-                .map(|(name, value)| leo_ast::CircuitImpliedVariableDefinition {
+                .map(|(name, value)| leo_ast::CircuitVarInit {
```

Comment:
Please change these back to the full `CircuitImpliedVariableDefinition`
While it is more verbose I think it is more clear.

Reply:
I'd like to stick with `CircuitVarInit`. Verbose is not necessarily more clear, and in this case, I think that applies.

Without looking at the definition and use of `CircuitImpliedVariableDefinition`, it was actually unclear to me what the type is for. To start with, I don't understand what `Implied` means in this context. Meanwhile, `Definition` sounds like it is refers to the place where a circuit is *defined*, not initialized.

The surrounding concept is aptly named `CircuitInitExpression`. And here we are initializing a fragment of that, namely a "variable" of the circuit, so `CircuitVariableInitializer` or `CircuitVarInit` using commonly shorthands seems apt to me.

Eventually, and once we `s/Circuit/Struct/` I'd like this to be called `StructFieldInit` since "variable" is more vague and "field" is more precise.

Reply:
Thank you for clarifying. Could we go for `CircuitVariableInitializer`?